### PR TITLE
Update javadocs for generic types in HttpServerTracer

### LINKED ISSUE #791: Document the STORAGE parameter in HttpServerTracer
It is confusing when/if you need it, e.g. https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/756#discussion_r459842332

Suggest to use `Void` if you do not need the `STORAGE` parameter for propagating the context to async handlers

### PR DESCRIPTION
**Pull request by breedx-splk**
This resolves #791

### CODE REVIEW

**Review by jkwatson — APPROVED** [review]
seems good to me. thanks!

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpServerTracer.java [hunk 1]
```diff
@@ -18,6 +18,19 @@
...
+
+/**
+ * Base class for implementing Tracers for HTTP servers. It has 3 types that must be specified by
+ * subclasses:
+ *
+ * @param <REQUEST> - The specific type for HTTP requests
+ * @param <RESPONSE> - The specific type for HTTP responses
+ * @param <CONNECTION> - The specific type of HTTP connection, used to get peer address information
+ *     and HTTP flavor. Use Void if your subclass does not need the connection for peerHostIP,
+ *     peerPort, or flavor.
```
Comment (trask) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpServerTracer.java:30]: can you check if there any subclasses that don't use CONNECTION, I'm not sure if makes sense to use Void for this (unlike STORAGE)

**Review by iNikem — APPROVED** [review]

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpServerTracer.java [hunk 2]
```diff
@@ -18,6 +18,19 @@
...
+ * @param <STORAGE> - Implementation specific storage type for attaching/getting the server context.
```
Comment (anuraaga) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpServerTracer.java:31]: Any doc is better than no doc so this LGTM, but maybe we can go further. `STORAGE` has always been very hard for me to grok. This comment still doesn't really tell me how to use this parameter, "what's server context", "what's implementation specific storage" are still remaining questions.

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpServerTracer.java (hunk 1 above)
Comment (breedx-splk) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpServerTracer.java:30]: > can you check if there any subclasses that don't use CONNECTION, I'm not sure if makes sense to use Void for this (unlike STORAGE)

Good call. They all do.  I'll remove the `Void` bit.

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpServerTracer.java (hunk 2 above)
Comment (trask) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpServerTracer.java:30]: Yeah, I'm planning to work on cleaning up the http server instrumentations after getting through the http clients, and will think on this.

**Review by trask — APPROVED** [review]

**Pull request merged**